File: algorithms/easy/minimum_number_game.ex
defmodule Solution do
  @spec number_game(nums :: [integer]) :: [integer]
  def number_game(nums) do
    nums
    |> Enum.sort()
    |> Enum.chunk_every(2)
    |> Enum.flat_map(fn [a, b] -> [b, a] end)
  end
end

File: algorithms/easy/minimum_number_game.rs
impl Solution {
    pub fn number_game(mut nums: Vec<i32>) -> Vec<i32> {
        nums.sort_unstable();
        for i in (0..nums.len()).step_by(2) {
            nums.swap(i, i + 1);
        }
        nums
    }
}
